### ISSUE TITLE
D:File Download Implementation

### ISSUE THREAD

**Issue by mhafidi** [mhafidi/3GPPWordSeriesAssembler#5]
Develop the logic to identify and download ZIP files from the FTP server.

**Comment by Melbatty** [comment]
Branch creation: gpp_dowi

**Comment by Melbatty** [comment]
Certainly! To develop the logic for identifying and downloading ZIP files from the FTP server using Java and Apache Commons Net, you can follow these steps:

Connect to the FTP Server:
Use Apache Commons Net to establish a connection to the FTP server.
java
Copy code
import org.apache.commons.net.ftp.FTP;
import org.apache.commons.net.ftp.FTPClient;

FTPClient ftpClient = new FTPClient();
ftpClient.connect("ftp.server.com");
ftpClient.login("username", "password");
ftpClient.enterLocalPassiveMode();
ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
Navigate to the Specified Directory:
Change the working directory to the target directory where the ZIP files are located.
java
Copy code
ftpClient.changeWorkingDirectory("/Specs/2023-09/Rel-17/");
List Files and Identify ZIP Files:
Retrieve a list of files in the current directory and identify ZIP files.
java
Copy code
FTPFile[] files = ftpClient.listFiles();
for (FTPFile file : files) {
    if (file.isFile() && file.getName().toLowerCase().endsWith(".zip")) {
        String zipFileName = file.getName();
        // Logic to download the ZIP file
        downloadZipFile(ftpClient, zipFileName);
    }
}
Download ZIP Files:
Implement the logic to download the identified ZIP files.
java
Copy code
private static void downloadZipFile(FTPClient ftpClient, String zipFileName) {
    try (OutputStream outputStream = new BufferedOutputStream(new FileOutputStream(zipFileName))) {
        boolean success = ftpClient.retrieveFile(zipFileName, outputStream);
        if (success) {
            System.out.println("File " + zipFileName + " downloaded successfully.");
        } else {
            System.err.println("Failed to download file " + zipFileName);
        }
    } catch (IOException e) {
        e.printStackTrace();
    }
}
Disconnect from the FTP Server:
Ensure to disconnect from the FTP server after completing the download.
java
Copy code
ftpClient.logout();
ftpClient.disconnect();
Remember to handle exceptions appropriately, and consider using try-with-resources for resource management. This logic assumes that you have already established a connection to the FTP server. Customize the code based on your specific requirements and error handling needs.
